scripts/run-modularization-gate.py

```
@@ -0,0 +1,168 @@
+        return sum(1 for _ in handle)
+
+
+def summarize_group(title: str, entries: list[str]) -> None:
+    if not entries:
+        return
+    print(f"[modularization-gate] {title} ({len(entries)})")
+    for entry in entries:
+        print(f"  - {entry}")
+
+
+def main() -> int:
```

Ruff (0.15.11)

[warning] Too many branches (19 > 12)

(PLR0912)